src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,48 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
+
+  checkIfAlive() {
+    if (this.health <= 0) {
+      Animal.alive = Animal.alive.filter((animal) => animal !== this);
+    }
+  }
```

Comment:
[CODE PERFORMANCE] - The `checkIfAlive` method uses a filter inside a loop (called every time an animal's health is checked), which can be inefficient. Consider filtering the `Animal.alive` array directly inside the `bite` method after reducing the health of the herbivore, instead of having a separate method that gets called every time an animal's health changes.